### PR TITLE
basic awesomebar

### PR DESCRIPTION
**Pull request by paulrouget**
This basic awesomebar implements Gordon's design. To feed the awesomebar, I use duckduckgo search completion and alexa top 500 list. This is a temporary solution, but it's good enough. There are some bugs related to input component and the tabstrip that we will fix later.

### CODE REVIEW

**Comment by paulrouget** [comment]
@Gozala r?

File: src/browser/awesomebar.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      suggestionsCursor.set('history', fromJS([]));
+      return;
+    }
+
+    xhrSearch = new XMLHttpRequest({mozSystem: true});
+    xhrSearch.open('GET', `https://ac.duckduckgo.com/ac/?q=${textInput}&type=list`, true);
+    xhrSearch.send();
+    xhrSearch.onload = () => {
+      try {
+        const json = JSON.parse(xhrSearch.response);
```
Comment (fabricedesre) [inline src/browser/awesomebar.js]: set xhr.responseType = 'json' to save you this parsing.

File: src/browser/awesomebar.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  let xhrHistory;
+
+  const computeSuggestions = (textInput, suggestionsCursor) => {
+
+    if (xhrSearch) xhrSearch.abort();
+    if (xhrHistory) xhrHistory.abort();
+
+    textInput = textInput.trim();
+
+    if (!textInput) {
```
Comment (Gozala) [inline src/browser/awesomebar.js]: Would be better do `suggestionsCursor.merge({search: List(), history: List()})` as that would not trigger two updates.

File: src/browser/awesomebar.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    let historyCount = Math.min(history.length,
+                                MAX_RESULTS - Math.min(MAX_RESULTS / 2, search.length));
+
+    for (let i = 0; i < searchCount; i++) {
+      let href = search[i].get('href');
+      let text = search[i].get('text');
+      suggestions.push(DOM.p({
+        className: 'suggestion search',
+        key: 'suggestionSearch' + i,
+        onMouseDown: e => onOpen(href)
```
Comment (Gozala) [inline src/browser/awesomebar.js]: Nit: Maybe worth putting an event handler onto just parent node.

File: src/browser/navigation-panel.js [hunk 4]
```diff
@@ -64,9 +65,18 @@ define((require, exports, module) => {
...
+        onFocus: event => {
+          computeSuggestions(event.target.value, suggestionsCursor);
+          inputCursor.set('isFocused', true);
+        },
+        onBlur: event => {
+          resetSuggestions(suggestionsCursor);
+          inputCursor.set('isFocused', false)
+        },
+        onChange: event => {
+          computeSuggestions(event.target.value, suggestionsCursor);
```
Comment (Gozala) [inline src/browser/navigation-panel.js]: It maybe worth to `require('lang/functional').throttle(computeSuggestion)` to reduce number of computations.

**Comment by Gozala** [comment]
Looks good to me!

**Pull request merged**